tests/test_data_sync_client.py

```diff
@@ -54,7 +55,7 @@ async def test_binary_data_capture_upload(self, service: MockDataSync):
                 data_request_times=DATETIMES,
                 binary_data=BINARY_DATA
             )
-            self.assert_sensor_contents(sensor_contents=service.sensor_contents, is_binary=True)
+            self.assert_sensor_contents(sensor_contents=list(service.sensor_contents), is_binary=True)
```

Comment:
I had to cast them to a list like this to avoid a linting error. I just got rid of the casting list(), though, and instead changed the type of the sensor_contents parameter of the assert_sensor_contents from List[SensorData] to a more broad Iterable[SensorData]. My guess is this is because the request definition only specifies it as an iterable, so python probably thinks we might not be getting back a list even though that's what I pass to the iterable parameter when making the request. Fixed now though!

Reply:
I _believe_ this was resolved offline, so going to resolve this comment. Please let me know if you still want to chat about it :)